zinc/src/main/scala/sbt/internal/inc/MixedAnalyzingCompiler.scala

```diff
@@ -181,9 +191,16 @@ object MixedAnalyzingCompiler {
       incrementalCompilerOptions: IncOptions,
       extra: List[(String, String)]
   ): CompileConfiguration = {
-    val classpathHash = classpath map { x =>
-      FileHash.of(x, Stamper.forHash(x).hashCode)
+    val parallelClasspathHashing = classpath.toParArray.map { file =>
```

Comment:
That's not my experience. Parallel collections do speed things up and they're reliable. I can tell you hitting the cache is not slower, I've done measurements locally.

Reply:
Even for 10 jars, doing it in parallel is faster than doing it in serial.

Reply:
It might be worth comparing the performance on an SSD versus a mechanical disk. It depends on a lot of factors (and I've no idea how modern technology mitigates them), but parallelizing IO operations on a mechanical disk might actually slow it down.

Reply:
In my experience, IO-bound operations yield very good results when they're parallelized. I think there are old textbooks on parallel computing that claim so too, before the pre-SSD era.
  
In any case, I benchmarked it just in case in a LAMP server. Results:
  
* Parallelized hashing: 549ms.
* Serial hashing: 1912ms.
  
I'm testing it on `/dev/vda`, which as you see is a rotational disk:
  
```
platform@scalagesrv2:~/tests/zinc$ lsblk -d -o name,rota
NAME ROTA
fd0     1
sr0     1
vda     1
```